From 18f4e3bf1579106f3cb9efb02ea2c9ea3e218ccc Mon Sep 17 00:00:00 2001
From: Lisa Treweek <ltreweek@altera.com>
Date: Thu, 14 Mar 2013 18:38:32 -0500
Subject: [PATCH 060/248] FogBugz #101898: device node tree corrected for
 correct flash part

Added the correct flash part number to the device tree.

[Original patch taken from
git://git.rocketboards.org/linux-socfpga.git socfpga-3.14]

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
Signed-off-by: Lisa Treweek <ltreweek@altera.com>
---
 arch/arm/boot/dts/socfpga_cyclone5.dtsi | 5 +++--
 drivers/mtd/devices/m25p80.c            | 1 +
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/socfpga_cyclone5.dtsi b/arch/arm/boot/dts/socfpga_cyclone5.dtsi
index 3082a29..3d0c8f1 100644
--- a/arch/arm/boot/dts/socfpga_cyclone5.dtsi
+++ b/arch/arm/boot/dts/socfpga_cyclone5.dtsi
@@ -101,10 +101,10 @@
 				fifo-depth = <128>;
 				bus-num = <2>;
 
-				flash0: n25q128@0 {
+				flash0: n25q00@0 {
 					#address-cells = <1>;
 					#size-cells = <1>;
-					compatible = "n25q128";
+					compatible = "n25q00";
 					reg = <0>;	/* chip select */
 					spi-max-frequency = <100000000>;
 					page-size = <256>;
@@ -147,6 +147,7 @@
 						reg = <0x0 0x1000000>;
 					};
 				};
+		};
 
 		i2c0: i2c@ffc04000 {
 			speed-mode = <0>;
diff --git a/drivers/mtd/devices/m25p80.c b/drivers/mtd/devices/m25p80.c
index 6e0d7f0..cc0c2a9 100644
--- a/drivers/mtd/devices/m25p80.c
+++ b/drivers/mtd/devices/m25p80.c
@@ -954,6 +954,7 @@ static const struct spi_device_id m25p_ids[] = {
 	{ "n25q128a13",  INFO(0x20ba18, 0, 64 * 1024,  256, 0) },
 	{ "n25q256a",    INFO(0x20ba19, 0, 64 * 1024,  512, SECT_4K) },
 	{ "n25q512a",    INFO(0x20bb20, 0, 64 * 1024, 1024, SECT_4K) },
+	{ "n25q00",   INFO(0x20ba21, 0, 64 * 1024, 256, SECT_4K) },
 
 	/* PMC */
 	{ "pm25lv512",   INFO(0,        0, 32 * 1024,    2, SECT_4K_PMC) },
-- 
1.9.1

